fire/cli/niv/_udtræk_revision.py

```diff
@@ -1,30 +1,248 @@
+    "SKITSE:png_md5",
+    "SKITSE:png_sti",
+    "ATTR:fundamentalpunkt",
+    "ATTR:tinglysningsnr",
+]
+"""
+Disse attributter indgår ikke i punktrevisionen
+(men det diskvalificerer ikke et punkt at have dem)
+"""
+
+ATTRIBUT_PRIORITERING: Final[list[str]] = [
+    "ATTR:muligt_datumstabil",
+    "ATTR:beskrivelse",
+]
+"""
+Attributter, der, hvis de eksisterer for et punkt, skal
+være øverst og i angivne rækefølge under hvert punkt.
+"""
+
+LOKATION_DEFAULT: Final[tuple[float, float]] = (11.0, 56.0)
+
+
+def hent_punkter_i_opmålingsdistrikter(
+    opmålingsdistrikter: list[str],
+) -> list:
+    if not opmålingsdistrikter:
+        return []
+
+    distrikter = ",".join([f"'{d.upper()}'" for d in opmålingsdistrikter])
+    uønsket = ",".join([f"'{p}'" for p in ATTRIBUTTER_UØNSKEDE])
+    pkt_i_distrikter = f"""
+        SELECT p.*
+        FROM (
+            SELECT DISTINCT g.punktid FROM geometriobjekt g
+            JOIN herredsogn hs
+            ON sdo_inside(g.geometri, SDO_GEOM.SDO_BUFFER(hs.geometri, 50, 0.1)) = 'TRUE'
+            WHERE
+                upper(hs.kode) IN ({distrikter})
+            AND
+                g.registreringtil IS NULL
+        ) a
+        LEFT JOIN (
+            SELECT DISTINCT pi.punktid FROM punktinfo pi
+            JOIN punktinfotype pit ON pit.infotypeid=pi.infotypeid
+            WHERE pit.infotype IN ({uønsket}) AND pi.registreringtil IS NULL
+        ) b
+        ON a.punktid = b.punktid
+        JOIN punkt p ON p.id = a.punktid
+        WHERE b.punktid IS NULL AND p.registreringtil IS NULL
+        ORDER BY p.registreringfra
+    """
+    stmt = text(pkt_i_distrikter).columns(Punkt.objektid)
+    return fire.cli.firedb.session.query(Punkt).from_statement(stmt).all()
+
+
+def lokations_streng(lokation: tuple[float, float]) -> str:
+    """
+    Returnerer tekst-repræsentation af lokationskoordinat.
+
+    Centimeterafrunding for lokationskoordinaten er rigeligt.
+
+    """
+    return f"{lokation[1]:.3f} m   {lokation[0]:.3f} m"
+
 
 
-@niv.command()
+def punkt_informationer_aktive(punkt_informationer: list[PunktInformation]) -> list[PunktInformation]:
+    return [
+        info
+        for info in punkt_informationer
+        if info.registreringtil is None
+    ]
+
+
+def fjern_ignorerede(punkt_informationer: list[PunktInformation], ignorerede: list[str]) -> list[PunktInformation]:
+    return [
+        info
+        for info in punkt_informationer
+        if info.infotype.name not in ignorerede
+    ]
+
+
+def har_landsnr_lig_ident(info: PunktInformation, ident: str) -> bool:
+    """Vis kun landsnr for punkter med GM/GI/GNSS-primærident"""
+    return info.infotype.name == "IDENT:landsnr" and info.tekst == ident
+
+
+def fjern_alle_med_ident(punkt_informationer: list[PunktInformation], ident: str) -> list[PunktInformation]:
+    return [
+        info
+        for info in punkt_informationer
+        if not har_landsnr_lig_ident(info, ident)
+    ]
+
+
+def flyt_attributter_til_toppen(
```

Comment:
Jeg tror dette kan gøres meget kortere ved at lade sort klare ærterne gennem en klassifikationsfunktion, der returnerer prioriteringsniveauet (eller 1 milliard for de uprioriterede) a la:

```python
# A function that returns the length of the value:
def myFunc(e):
  return len(e)

cars = ['Ford', 'Mitsubishi', 'BMW', 'VW']

cars.sort(key=myFunc)
```